src/Atmos/Model/prog_prim_conversion.jl

```diff
@@ -93,14 +107,10 @@ function prognostic_to_primitive!(
     moist::NonEquilMoist,
     prim::Vars,
     prog::Vars,
-    e_int::AbstractFloat,
+    aux::Vars,
 )
-    q_pt = PhasePartition(
-        prog.moisture.ρq_tot / prog.ρ,
-        prog.moisture.ρq_liq / prog.ρ,
-        prog.moisture.ρq_ice / prog.ρ,
-    )
-    ts = PhaseNonEquil(atmos.param_set, e_int, prog.ρ, q_pt)
+    ts = recover_thermo_state(atmos, prog, aux)
```

Comment:
I think this is wrong. `recover_thermo_state` is intended to assume that `aux.T` exists and is correctly populated.